src/math/WadRayMath.sol

```diff
@@ -30,7 +30,7 @@ library WadRayMath {
         // <=> x * y > type(uint256).max - HALF_WAD
         // <=> x > (type(uint256).max - HALF_WAD) / y
         assembly {
-            if and(gt(y, 0), gt(x, div(MAX_UINT256_MINUS_HALF_WAD, y))) {
+            if mul(y, gt(x, div(MAX_UINT256_MINUS_HALF_WAD, y))) {
```

Comment:
To be precise we can add the condition `y != 0` in the last line in the comment above

Reply:
`y` is already declared positive L28

```solidity
// Let y > 0
```

Reply:
True, I missed that. But in the case of multiplication it's not clear why we would assume that at the start. In contrast, the last equivalence makes it clear why we are checking this condition: we are doing a division by `y`